lib/rucio/core/rule.py

```diff
@@ -254,28 +254,69 @@ def add_rule(dids, account, copies, rse_expression, grouping, weight, lifetime,
 
             if ask_approval:
                 new_rule.state = RuleState.WAITING_APPROVAL
-                # Block manual approval for multi-rse rules
-                if len(rses) > 1:
-                    raise InvalidReplicationRule('Ask approval is not allowed for rules with multiple RSEs')
-                if len(rses) == 1 and not did.is_open and did.bytes is not None and did.length is not None:
-                    # This rule can be considered for auto-approval:
-                    rse_attr = list_rse_attributes(rse_id=rses[0]['id'], session=session)
-                    auto_approve = False
-                    if 'auto_approve_bytes' in rse_attr and 'auto_approve_files' in rse_attr:
-                        if did.bytes < int(rse_attr.get('auto_approve_bytes')) and did.length < int(rse_attr.get('auto_approve_bytes')):
-                            auto_approve = True
-                    elif did.bytes < int(rse_attr.get('auto_approve_bytes', -1)):
-                        auto_approve = True
-                    elif did.length < int(rse_attr.get('auto_approve_files', -1)):
+                configured_algorithm, auto_approve = None, False
```

Comment:
I think this setup part should really be on root of the module. Otherwise you are reloading the modules every time? @jamesp-epcc I think this is also how it is done in other algorithms?

Reply:
Hi @bari12 
It will not cause any more reloads than when defined in the root of the module. 
The module will be loaded in the `sys.modules` it will remain there till the module's lifetime.